presto-main/src/main/java/com/facebook/presto/operator/aggregation/FixedDoubleHistogram.java

```diff
@@ -0,0 +1,184 @@
+    }
+
+    FixedDoubleHistogram(FixedDoubleHistogram other)
+    {
+        this(other.bucketCount, other.min, other.max);
+        for (int i = 0; i < bucketCount; ++i) {
+            weights[i] = other.weights[i];
+        }
+    }
+
+    public int getRequiredBytesForSerialization()
+    {
+        return SizeOf.SIZE_OF_BYTE + // format
+                SizeOf.SIZE_OF_INT + // num buckets
+                SizeOf.SIZE_OF_DOUBLE + // min
+                SizeOf.SIZE_OF_DOUBLE + // max
+                SizeOf.SIZE_OF_DOUBLE * bucketCount; // weights
+    }
+
+    public void serialize(SliceOutput out)
+    {
+        out.appendByte(FORMAT_TAG)
+            .appendInt(bucketCount)
+            .appendDouble(min)
+            .appendDouble(max)
+            .appendBytes(Slices.wrappedDoubleArray(weights, 0, bucketCount));
+    }
+
+    public long estimatedInMemorySize()
+    {
+        return INSTANCE_SIZE + SizeOf.sizeOf(weights);
+    }
+
+    public void set(double value, double weight)
+    {
+        checkArgument(
+                value >= min && value <= max,
+                "value must be within range [%s, %s]", min, max);
+        weights[getIndexForValue(value)] = weight;
+    }
+
+    public void add(double value)
+    {
+        add(value, 1);
+    }
+
+    public void add(double value, double weight)
+    {
+        weights[getIndexForValue(value)] += weight;
+    }
+
+    public int getBucketCount()
+    {
+        return bucketCount;
+    }
+
+    public void mergeWith(FixedDoubleHistogram other)
+    {
+        checkArgument(min == other.min, "min %s must be equal to other min %s", min, other.min);
+        checkArgument(max == other.max, "max %s must be equal to other max %s", max, other.max);
+        checkArgument(
+                bucketCount == other.bucketCount,
+                "bucketCount %s must be equal to other bucketCount %s", bucketCount, other.bucketCount);
+
+        for (int i = 0; i < bucketCount; ++i) {
+            weights[i] += other.weights[i];
+        }
+    }
+
+    @Override
+    public Iterator<Bin> iterator()
+    {
+        return new Iterator<Bin>()
+        {
+            private int currentIndex;
+
+            @Override
+            public boolean hasNext()
+            {
+                return currentIndex < bucketCount;
+            }
+
+            @Override
+            public Bin next()
+            {
+                if (currentIndex == bucketCount) {
+                    throw new UnsupportedOperationException();
+                }
+
+                return new Bin(
+                    currentIndex * (max - min) / bucketCount,
+                    (currentIndex + 1) * (max - min) / bucketCount,
+                    weights[currentIndex++]);
+            }
+
+            @Override
+            public void remove()
+            {
+                throw new UnsupportedOperationException();
+            }
+        };
+    }
+
+    private int getIndexForValue(Double value)
```

Comment:
I'm fine with the tests having big-D doubles but let's remove all of them from main, we don't need the autoboxing.

Reply:
Thanks for pointing this out - am removing them from the tests as well.